### ISSUE TITLE
Clang linker error while compiling with AddressSanitizer

### ISSUE THREAD

**Issue by vish99** [google/sanitizers#1188]
I installed Clang along with LLVM. I'm trying to understand ASAN. When I try running clang along with ASAN flag like:

`clang -fsanitize=address <file_name>`

I get the following error
`ld: file not found: /Users/vishal/Desktop/llvm/llvmproject/build/lib/clang/11.0.0/lib/darwin/libclang_rt.asan_osx_dynamic.dylib
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)`

Clang Version:
`clang version 11.0.0 (https://github.com/llvm/llvm-project.git b780df052dd2b246a760d00e00f7de9ebdab9d09)
Target: x86_64-apple-darwin18.0.0
Thread model: posix
InstalledDir: /Users/vishal/Desktop/llvm/llvm-project/build/bin/.`

I see it's an issue with the linker. ASAN instrumented LLVM IR is being produced without any error. The issue seems to be linking the ASAN run-time library using Clang. Any help would be appreciated.

**Comment by mas2tg** [comment]
Hey, I was having a similar issue the other day (yesterday) and I was able to compile my program in my Mac using the next flags. Let me know if this works for you!

-L/Library/Developer/CommandLineTools/usr/lib/clang/11.0.0/lib/darwin 
-lSystem /Library/Developer/CommandLineTools/usr/lib/clang/11.0.0/lib/darwin/libclang_rt.asan_osx_dynamic.dylib

**Comment by vish99** [comment]
No, it didn't work for me. Actually there's no directory named "11.0.0" inside `/Library/Developer/CommandLineTools/usr/lib/clang/`. Insted there exists "10.0.1" directory, which I guess is Apple Clang.
  
I want to change the code of AddressSanitizer to do the instrumentation differently. So I'll need to deal with the AddressSanitizer file built out from LLVM code base and not the Apple Clang one.
